src/app/src/test/kotlin/org/vaccineimpact/api/tests/logic/CoverageLogic/SaveCoverageTests.kt

```diff
@@ -172,37 +175,42 @@ and 13 others"""
     @Test
     fun `validation detects bad campaign years`()
     {
-        val mockRepo = mock<TouchstoneRepository> {
-            on { getGenders() } doReturn mapOf(GenderEnum.BOTH to 111)
-        }
-        val testSequence = sequenceOf(
-                CoverageIngestionRow("HepB", "AFG", ActivityType.CAMPAIGN, true, 2010, 1, 10, GenderEnum.BOTH, 100F, 65.5F, false)
-        )
-        val sut = RepositoriesCoverageLogic(mock(), mock(), mockRepo, mock(), mockExpectationsRepo)
-        val expectedMessage = "Unexpected year: 2010"
-
-        assertThatThrownBy {
-            sut.saveCoverageForTouchstone("t1", testSequence, "", "", now)
-        }.isInstanceOf(BadRequest::class.java)
-                .hasMessageContaining(expectedMessage)
+       `validation detects bad years`(ActivityType.CAMPAIGN)
     }
 
     @Test
     fun `validation detects bad routine years`()
+    {
+        `validation detects bad years`(ActivityType.ROUTINE)
+    }
+
+    private fun `validation detects bad years`(activityType: ActivityType)
     {
         val mockRepo = mock<TouchstoneRepository> {
             on { getGenders() } doReturn mapOf(GenderEnum.BOTH to 111)
         }
-        val testSequence = sequenceOf(
-                CoverageIngestionRow("HepB", "AFG", ActivityType.ROUTINE, true, 2010, 1, 10, GenderEnum.BOTH, 100F, 65.5F, false)
-        )
+
         val sut = RepositoriesCoverageLogic(mock(), mock(), mockRepo, mock(), mockExpectationsRepo)
-        val expectedMessage = "Unexpected year: 2010"
 
+        val testSequenceEarly = sequenceOf(
+                CoverageIngestionRow("HepB", "AFG", activityType, true, 1999, 1, 10, GenderEnum.BOTH, 100F, 65.5F, false)
+        )
+
+        val testSequenceLate = sequenceOf(
+                CoverageIngestionRow("HepB", "AFG", activityType, true, 21019, 1, 10, GenderEnum.BOTH, 100F, 65.5F, false)
```

Comment:
`21019` _looks_ a bit odd, given that the assertion on L209 is expecting `2101` - but I might be missing something?!

Reply:
Oops! Yeah, you're right, that should have failed the test. I'll have look.

Reply:
Ha, this actually passed the test because it was only checking for a partial message since they whole thing is a bit long-winded - and "Unexpected year: 2101" is indeed a substring of "Unexpected year: 21019... etc" :facepalm: